File: code/jvm/im-http-api/src/main/kotlin/im/api/model/output/users/UsersPaginatedOutputModel.kt
package im.api.model.output.users

import im.api.model.output.PaginationOutputModel
import im.domain.user.User
import im.repository.pagination.Pagination

data class UsersPaginatedOutputModel(
    val users: List<UserOutputModel>,
    val pagination: PaginationOutputModel?,
) {
    companion object {
        fun fromDomain(users: Pagination<User>): UsersPaginatedOutputModel =
            UsersPaginatedOutputModel(
                users = users.items.map { UserOutputModel.fromDomain(it) },
                pagination = PaginationOutputModel.fromPagination(users.info),
            )
    }
}

File: code/client/web/src/Dto/output/users/UsersPaginatedOutputModel.ts
import { PaginationOutputModel } from '../PaginationOutputModel';
import { UserOutputModel } from './UserOutputModel';

/**
 * Output model for paginated users
 *
 * @param users The users in the current page.
 * @param pagination The pagination information.
 */
export interface UsersPaginatedOutputModel {
    users: UserOutputModel[];
    pagination: PaginationOutputModel;
}
